design/external-authorization-http-support-design.md

```diff
@@ -0,0 +1,165 @@
+# Global HTTP External Authorization
+
+Status: Draft
+
+## Abstract
+
+HTTP Support for External Authorization.
+
+## Background
+See [#4954](https://github.com/projectcontour/contour/issues/4954) for context.
+
+## Goals
+- Support Envoy's L7 External Authorization filter for HTTP Virtual Hosts.
+- Extend the functionality of existing External Authorization.
+
+## Non Goals
+- Supporting more than one External Authorization Configuration for HTTP endpoints.
+
+## High-Level Design
+
+Contour will add HTTP support for Envoy's External Authorozation.
+
+Currently, Contour supports External Authorization only over HTTPS. External Authorization allows for requests to be authenticated before they are forwarded upstream. However, External authorization over HTTPS is incompatible with setups that terminate TLS at the cloud provider's load balancer layer because the Envoy HCM relies on the SNI to choose the correct filter chain. 
+
+A new type, `GlobalHTTPExtAuth`, will be defined as part of the `ContourConfiguration`.
+A `GlobalHTTPExtAuth` config would define a global external authorization configuration for all the HTTP routes. 
+
+## Detailed Design
+
+### Contour Configuration changes
+By using global external authorization, an external authorization service can be configured in the Contour config file.
+This External authorization configuration will be used for all HTTP routes. 
+
+```yaml
+globalHTTPExtAuth:
+  # extensionService is the <namespace>/<name>
+  # of the ExtensionService we created in the
+  # previous step.
+  extensionService: projectcontour-auth/htpasswd
+  # failOpen is whether to allow requests through
+  # if there's an error.
+  failOpen: false
+  # Context is a set of key/value pairs that are sent to the authentication server in the check request.
+  context:
+    header1: value1
+    header2: value2
+    # ResponseTimeout configures maximum time to wait for a check response from the authorization server
+  responseTimeout: 1s
+
+```
+
+### HTTPS Override
+If an upstream has HTTPS enabled, the HTTPS configuration will take precedence. 
```

Comment:
is this basically because they will be on different filters? we should probably follow the precedent of the project. 

Does contour have any other configurations that can be applied both with `configmap` and with `annotation`?

Reply:
we do have precedent for doing some global configuration with a per-route override, we allow configuration of reqest/response header modification at the [config file](https://projectcontour.io/docs/v1.23.2/configuration/#policy-configuration) and [per-route](https://projectcontour.io/docs/v1.23.2/config/api/#projectcontour.io/v1.Route), with the aim being the former is configured by the persona operating the contour installation, and the latter for any app-operator/developer specific overrides

when we support this feature in Gateway API seems like there may be room here to implement this as a Policy resource and use the override capabilities there